### PR TITLE
rename Network to BitcoinNetwork create new Network interface (improved PR)

### PR DESCRIPTION
**Pull request by msgilligan**
This change give us flexibility and preservers the useful abstraction that we currently have with `NetworkParameters`.

This PR replaces PR #2473. It has some minor improvements, but more importantly it contains two commits which preservers the file history of `Network` -> `BitcoinNetwork`.

### CODE REVIEW

**Comment by msgilligan** [comment]
Making this a DRAFT PR. We'll wait until after the current set of refactorings and revisit this issue.

File: core/src/main/java/org/bitcoinj/utils/Network.java [hunk 1]
```diff
@@ -16,54 +16,18 @@
...
-     */
-    public static Network of(String idString) {
-        return Arrays.stream(values())
-                .filter(n -> n.id.equals(idString))
-                .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException("Illegal network ID: " + idString));
-    }
-}
+    String id();
+}
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/utils/Network.java:33]: No newline at end of file

**Comment by schildbach** [comment]
Would you like to rebase this?

**Comment by schildbach** [comment]
Concept looks good.

**Comment by msgilligan** [comment]
> Would you like to rebase this?

I re-implemented. See PR #2501.

**Pull request closed without merging**